packages/@aws-cdk/aws-appsync/lib/graphqlapi.ts

```diff
@@ -1027,6 +1066,128 @@ export abstract class MappingTemplate {
+   *
+   * @param resourcePath e.g. if full path is https://api.xxxxxxxxx.com/posts then resourcePath would be /posts
+   */
+  public static httpPostAndForwardArgumentsAsJson(resourcePath: string = '/'): MappingTemplate {
+    return this.fromString(`{
+      "version": "2018-05-29",
+      "method": "POST",
+      "resourcePath": "${resourcePath}",
+      "params":{
+          "body": $util.toJson($ctx.args),
+          "headers":{
+              "Content-Type": "application/json",
+              "Authorization": "$ctx.request.headers.Authorization"
+          }
+      }
+    }`);
+  }
+
+  /**
+   * HTTP Request Mapping Template Sample: GET and forward headers
+   *
+   * @param resourcePath e.g. if full path is https://api.xxxxxxxxx.com/posts then resourcePath would be /posts
+   */
+  public static httpGetAndForwardHeaders(resourcePath: string = '/'): MappingTemplate {
+    return this.fromString(`{
+      "version": "2018-05-29",
+      "method": "GET",
+      "resourcePath": "${resourcePath}",
+      "params":{
+          ## you can forward the headers using the below utility
+          "headers": $utils.http.copyheaders($ctx.request.headers)
+      }
+    }`);
+  }
+
+  /**
+   * HTTP Request Mapping Template Sample: Simple GET
+   *
+   * @param resourcePath e.g. if full path is https://api.xxxxxxxxx.com/posts then resourcePath would be /posts
+   */
+  public static httpSimpleGet(resourcePath: string = '/'): MappingTemplate {
+    return this.fromString(`{
+      "version": "2018-05-29",
+      "method": "GET",
+      "resourcePath": "${resourcePath}"
+    }`);
+  }
+
+  /**
+   * HTTP Request Mapping Template Sample: Simple DELETE
+   *
+   * @param resourcePath e.g. if full path is https://api.xxxxxxxxx.com/posts then resourcePath would be /posts
+   */
+  public static httpSimpleDelete(resourcePath: string = '/'): MappingTemplate {
+    return this.fromString(`{
+      "version": "2018-05-29",
+      "method": "DELETE",
+      "resourcePath": "${resourcePath}"
+    }`);
+  }
+
+  /**
+   * HTTP Response Mapping Template Sample: Return reponse on 200 (OK)
+   */
+  public static httpReturnResponseOn200Ok(): MappingTemplate {
+    return this.fromString(`
+      ## Raise a GraphQL field error in case of a datasource invocation error
+      #if($ctx.error)
+        $util.error($ctx.error.message, $ctx.error.type)
+      #end
+      ## if the response status code is not 200, then return an error. Else return the body **
+      #if($ctx.result.statusCode == 200)
+          ## If response is 200, return the body.
+          $ctx.result.body
+      #else
+          ## If response is not 200, append the response to error block.
+          $utils.appendError($ctx.result.body, "$ctx.result.statusCode")
+      #end
+    `);
+  }
+
+  /**
+   * HTTP Response Mapping Template Sample: Return XML reponse as json
+   */
+  public static httpReturnXmlResponseAsJson(): MappingTemplate {
```

Comment:
I think we should take out this method for now as well. I may be mistaken in thinking this wont be widely used, but its easier to add later if users ask for it then it is to remove if we decide we want to.